### PR TITLE
[ADD] web_view_searchpanel

### PR DESCRIPTION
**Pull request by etobella**
This PR is intended to be used by OCA/dms portback to 12.0. You can follow the conversation on OCA/dms#1

In order to test it you can do the following:
* Install `hr`
* Access employee kanban view in debug mode
* Edit the kanban view
* Add the following code on the kanban view
```
<searchpanel>
    <field name="department_id" select="one"/>
    <field name="job_id" select="multi" domain="[('department_id', '=', department_id)]"/>
</searchpanel>
```
* Refresh
image

@keshrath @pedrobaeza @agenterp @Muk-It

### CODE REVIEW

**Comment by pedrobaeza** [comment]
Please check Travis.

Can you please add a searchpanel as demo data in res.users using the field "user type" that serves for testing it?

**Comment by etobella** [comment]
Sure!! Added a demo only launched on demo. user type cannot be used, as it is not a real field. I have user `tz` and `lang`

File: web_view_searchpanel/__manifest__.py [hunk 1]
```diff
@@ -0,0 +1,27 @@
+# Copyright 2017-2019 MuK IT GmbH.
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl.html).
+
+{
+    'name': 'Search Panel',
+    'summary': 'Kanban Search Panel',
+    'version': '12.0.2.0.1',
```
Comment (pedrobaeza) [inline web_view_searchpanel/__manifest__.py]: Why this version?

**Review by pedrobaeza — APPROVED** [review]

File: web_view_searchpanel/template/assets.xml [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+		</xpath>
+		<xpath expr="//script[last()]" position="after">
+			<script type="text/javascript" src="/web_view_searchpanel/static/src/js/kanban_searchpanel.js" />
+			<script type="text/javascript" src="/web_view_searchpanel/static/src/js/kanban_controller.js" />
+			<script type="text/javascript" src="/web_view_searchpanel/static/src/js/kanban_renderer.js" />
+			<script type="text/javascript" src="/web_view_searchpanel/static/src/js/kanban_view.js" />
+		</xpath>
+	</template>
+
+</odoo>
```
Comment (marcelsavegnago) [inline web_view_searchpanel/template/assets.xml]:
```suggestion
</odoo>

```

File: web_view_searchpanel/demo/res_users_views.xml [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+            <xpath expr="//templates" position="after">
+                <searchpanel>
+                    <field name="lang" select="multi"/>
+                    <field name="tz" select="multi"/>
+                </searchpanel>
+            </xpath>
+        </field>
+    </record>
+
+</odoo>
```
Comment (marcelsavegnago) [inline web_view_searchpanel/demo/res_users_views.xml]:
```suggestion
</odoo>

```

**Review by marcelsavegnago — APPROVED** [review]

File: web_view_searchpanel/static/src/js/kanban_controller.js [hunk 4]
```diff
@@ -0,0 +1,58 @@
+/**
+*
+*    Copyright 2017-2019 MuK IT GmbH.
+*    License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl.html).
+*
+**/
+
+odoo.define('muk_web_searchpanel.KanbanController', function (require) {
```
Comment (legalsylvain) [inline web_view_searchpanel/static/src/js/kanban_controller.js]: nitpicking : could you replace ``muk_web_searchpanel`` by ``web_view_searchpanel`` ?

File: web_view_searchpanel/static/src/js/kanban_renderer.js [hunk 5]
```diff
@@ -0,0 +1,19 @@
+/**
+*
+*    Copyright 2017-2019 MuK IT GmbH.
+*    License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl.html).
+*
+**/
+
+odoo.define('muk_web_searchpanel.KanbanRenderer', function (require) {
```
Comment (legalsylvain) [inline web_view_searchpanel/static/src/js/kanban_renderer.js]: same here.

File: web_view_searchpanel/static/src/js/kanban_searchpanel.js [hunk 6]
```diff
@@ -0,0 +1,631 @@
+/**********************************************************************************
+*
+*    Copyright 2017-2019 MuK IT GmbH.
+*    License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl.html).
+*
+**********************************************************************************/
+
+odoo.define('web.SearchPanel', function (require) {
```
Comment (legalsylvain) [inline web_view_searchpanel/static/src/js/kanban_searchpanel.js]: the namespace should not be web_view_searchpanel.SearchPanel, here ?

File: web_view_searchpanel/static/src/js/kanban_view.js [hunk 7]
```diff
@@ -0,0 +1,118 @@
+/**
+*
+*    Copyright 2017-2019 MuK IT GmbH.
+*    License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl.html).
+*
+**/
+
+odoo.define('muk_web_searchpanel.KanbanView', function (require) {
```
Comment (legalsylvain) [inline web_view_searchpanel/static/src/js/kanban_view.js]: and same here.

**Review by legalsylvain — APPROVED** [review]
Hi. some minor remarks. 
otherwise, thanks a lot for backporting this feature ! 
tested on runbot, (with your exemple and hr).

Works like a charm. 
thanks !

**Comment by pedrobaeza** [comment]
s/web_biew_searchpanel/web_view_searchpanel

**Comment by etobella** [comment]
All comments attended!

**Review by jarroyomorales — APPROVED** [review]
LGTM! :+1:

File: web_view_searchpanel/static/src/js/kanban_controller.js [hunk 8]
```diff
@@ -0,0 +1,58 @@
+/**
+*
+*    Copyright 2017-2019 MuK IT GmbH.
+*    License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl.html).
+*
+**/
+
+odoo.define('web_biew_searchpanel.KanbanController', function (require) {
```
Comment (pedrobaeza) [inline web_view_searchpanel/static/src/js/kanban_controller.js]: and the same for the rest

```suggestion
odoo.define('web_view_searchpanel.KanbanController', function (require) {
```

Comment (etobella) [reply web_view_searchpanel/static/src/js/kanban_controller.js]: Sorry, I had the search with a defined kind of files and I missed them all :disappointed:

**Comment by pedrobaeza** [comment]
Not sure if this fix is needed here:

https://github.com/odoo/odoo/pull/56319

**Comment by AnaitzBusman** [comment]
how does this work?
i'm adding  the previous code on the kanban view an no searchpanel is displayed.

**Comment by etobella** [comment]
@AnaitzBusman Here you can see an example:
https://github.com/OCA/dms/blob/12.0/dms/views/directory.xml#L345

**Pull request merged**